app/models/manageiq/providers/vmware/infra_manager/refresher.rb

```diff
@@ -4,281 +4,18 @@
-            @vi.pbmProfilesByUid.reject { |_uid, profile| profile.profileCategory == "DATA_SERVICE_POLICY" }
-          end
-
-          unless @vc_data[:storage_profile].blank?
-            storage_profile_ids = @vc_data[:storage_profile].keys
-
-            collect_and_log_inventory(ems, :storage_profile_entity) { @vi.pbmQueryAssociatedEntity(storage_profile_ids) }
-
-            if targets.include?(ems)
-              collect_and_log_inventory(ems, :storage_profile_datastore) { @vi.pbmQueryMatchingHub(storage_profile_ids) }
-            end
-          end
-        end
-      end
-
-      def get_vc_data_ems_customization_specs(ems)
-        log_header = format_ems_for_logging(ems)
-
-        cleanup_callback = proc { @vc_data = nil }
-
-        retrieve_from_vc(ems, cleanup_callback) do
-          _log.info("#{log_header} Retrieving Customization Spec inventory...")
-          begin
-            vim_csm = @vi.getVimCustomizationSpecManager
-            @vc_data[:customization_specs] = vim_csm.getAllCustomizationSpecs
-          rescue RuntimeError => err
-            raise unless err.message.include?("not supported on this system")
-            _log.info("#{log_header} #{err}")
-          ensure
-            vim_csm.release if vim_csm rescue nil
-          end
-          _log.info("#{log_header} Retrieving Customization Spec inventory...Complete - Count: [#{@vc_data[:customization_specs].length}]")
-
-          EmsRefresh.log_inv_debug_trace(@vc_data[:customization_specs], "#{_log.prefix} #{log_header} customization_spec_inv:")
-        end
-      end
-
-      def get_vc_data_host_scsi(ems, host_mors)
-        log_header = format_ems_for_logging(ems)
-        return _log.info("#{log_header} Not retrieving Storage Device inventory for hosts...") if host_mors.empty?
-
-        cleanup_callback = proc { @vc_data = nil }
-
-        retrieve_from_vc(ems, cleanup_callback) do
-          _log.info("#{log_header} Retrieving Storage Device inventory for [#{host_mors.length}] hosts...")
-          host_mors.each do |mor|
-            data = @vc_data.fetch_path(:host, mor)
-            next if data.nil?
-
-            _log.info("#{log_header} Retrieving Storage Device inventory for Host [#{mor}]...")
-            begin
-              vim_host = @vi.getVimHostByMor(mor)
-              sd = vim_host.storageDevice(:ems_refresh_host_scsi)
-              data.store_path('config', 'storageDevice', sd.fetch_path('config', 'storageDevice')) unless sd.nil?
-            ensure
-              vim_host.release if vim_host rescue nil
-            end
-            _log.info("#{log_header} Retrieving Storage Device inventory for Host [#{mor}]...Complete")
-          end
-          _log.info("#{log_header} Retrieving Storage Device inventory for [#{host_mors.length}] hosts...Complete")
-
-          EmsRefresh.log_inv_debug_trace(@vc_data[:host], "#{_log.prefix} #{log_header} host_inv:")
-        end
+        # TODO remove from core
```

Comment:
Will a refresh from console "just work"?  If so then :tada:

Reply:
Yeah exactly, now the standard `EmsRefresh.refresh(ems)` will "just work" but use RbVmomi instead of MiqVim